src/Atlassian.Stash/Workers/HttpCommunicationWorker.cs

```diff
@@ -5,39 +5,40 @@
 using System.Net.Http.Headers;
 using System.Text;
 using System.Threading.Tasks;
+using Atlassian.Stash.Api.Enums;
 
 namespace Atlassian.Stash.Workers
 {
     internal class HttpCommunicationWorker
     {
         private Uri baseUrl;
         private AuthenticationHeaderValue authenticationHeader = null;
-
+		
         public HttpCommunicationWorker(string baseUrl, string base64Auth)
         {
             this.baseUrl = new Uri(baseUrl);
 
-            SetBasicAuthentication(base64Auth);
+            SetBasicAuthentication(base64Auth, AuthScheme.Bearer);
```

Comment:
The original implementation built around support for basic authentication only. 
`SetBasicAuthentication(string base64Auth)` and `SetBasicAuthentication(string username, string password)` only differ on the way credentials are provided: explicit username and password versus combination of `username:password` converted to base64.

Given the function name is `SetBasicAuthentication`, it might misleading if we provide Bearer Token support this way. Therefor I prefer a new method is created name `SetBearerTokenAuthentication` or similar. The logic of setting the authentication header value `this.authenticationHeader` can be moved to a private method and the all authentication method can call it.

Everything else is perfect.
Thanks again for your contribution.

Reply:
No worries, I didn't realise that was the intent of the base64 basic auth but looking at it now it's obvious! When I get a chance today I'll update the PR.